crates/cheatcodes/src/evm.rs

```diff
@@ -302,11 +303,11 @@ impl Cheatcode for storeCall {
 
 impl Cheatcode for coolCall {
     fn apply_full<DB: DatabaseExt>(&self, ccx: &mut CheatsCtxt<DB>) -> Result {
-        let Self { target } = self;
-        if let Some(account) = ccx.data.journaled_state.state.get_mut(target) {
-            account.unmark_touch();
-            account.storage.clear();
-        }
+        let Self { target } = *self;
+        ensure_not_precompile!(&target, ccx);
+        // TODO: prevent or warn about cooling the to/from address in a tx
+        ccx.state.addresses.insert(target, AddressState::Cool);
+        ccx.state.address_storage.insert(target, rHashMap::new());
```

Comment:
Can these be merged into one map?

Reply:
possibly? I wasn't sure how to do this, since i want to track the address state separately from the address's slots? I could make another struct that contains the state + the map?

Reply:
you can use a tuple for `(AddressState, HashMap...)`

Reply:
oh nice! is that better/easier than making a `pub struct CoolState { pub address: AddressState .. }`?

Reply:
It's more convenient since you're only using this struct in this one map and for one purpose

Reply:
ok makes sense, I just pushed with the struct changes but can convert it

Reply:
though `if cool_state.address == AddressState::Cool` reads better than `if cool_state.0 == AddressState::Cool` unless we're ok with that? i assume there's a way to name them too

Reply:
you can destructure tuples like anything else `let (a, b) = ...`, `if let Some((a, b)) = ...`

Reply:
cool, seems to work